dist/moment-range.js

```diff
@@ -337,6 +330,67 @@ DateRange.prototype.diff = function(unit) {
   return this.end.diff(this.start, unit);
 };
 
+DateRange.prototype.set = function(start, end) {
+  var parts;
+  var s = start;
+  var e = end;
+
+  if (arguments.length === 1 || end === undefined) {
+    if (typeof start === 'object' && start.length === 2) {
+      s = start[0];
+      e = start[1];
+    }
+    else if (typeof start === 'string') {
+      parts = start.split('/');
+      s = parts[0];
+      e = parts[1];
+    }
+  }
+  
+  if(this.options.limit !== false){
+    this.actualStart = (s === null) ? moment(-8640000000000000) : moment(s);
+    this.actualEnd   = (e === null) ? moment(8640000000000000) : moment(e);
+    this.actualRange = moment.range(this.actualStart, this.actualEnd, {limit:false});
+
+    this._intersect();
+  }
+  else {
+    this.start = (s === null) ? moment(-8640000000000000) : moment(s);
+    this.end   = (e === null) ? moment(8640000000000000) : moment(e);
+  }
+}
+
+DateRange.prototype.setLimits = function (options) {
+  this.lowerLimit = moment(-8640000000000000);
+  this.upperLimit = moment(8640000000000000);
+  this.limitRange = moment.range(this.lowerLimit, this.upperLimit, {limit: false});
+}
+
+DateRange.prototype._intersect = function() {
+  var limit = this.limitRange;
+  var actual = this.actualRange;
+
+  if ((actual.start <= limit.start) && (limit.start < actual.end) && (actual.end < limit.end)) {
```

Comment:
This whole chunk of code can be rewritten as 


```
this.start = actual.start >= limit.start ? actual.start : limit.start;
this.end = actual.end <= limit.end ? actual.end : limit.end;
```

If you want to still check for the range limits and throw the error then you can also add 

```
if(actual.end < limit.start || actual.start > limit.start) {
 throw ....
}

```